oceanprotocol/ocean-node
# Dashboard authentication for stop node & logs

Fixes #293 
Fixes: #357
Fixes: #296 
Fixes: #364
Fixes: #326

Changes proposed in this PR:

- Stop node command requires the signing of the message
- Logs require the signing of a message
- Both of the `/logs` and `/logs:id` endpoints are now post requests where the body of the request includes `expiryTimestamp` and `signature`
- Signature is stored in local storage
- `expiryTimestamp` is set to 12hrs from curent time. 
- Using rainbow kit for wallet connection
- Using MUI for easier styling of components
- Logs now send up to 90 days of logs (with a maximum of 100 logs) by default. 
- Refactoring of admin handlers

To Test:

1. make sure that you export `ALLOWED_ADMINS` and include your wallet address in the array.
2. `npm run build && npm run start` from the root of the directory 
3. navigate to `http://localhost:8000/dashboard`
4. connect wallet
5. Click "unlock" to sign a mesage which will authorise you to call the admin endpoints for 12hrs. 
6. Attempting to do the same with an account that isn't listed in the `ALLOWED_ADMINS`  array should not work. You should see a message saying "Your account does not have admin access"
